api/host_query_db.py

```diff
@@ -557,3 +564,44 @@ def get_host_ids_list(
     host_list = [str(res[0]) for res in _find_all_hosts(base_query, [Host.id]).filter(*all_filters).all()]
     db.session.close()
     return host_list
+
+
+def get_hosts_to_export(identity: object, filters: object = {}, export_format: str = "json", rbac_filter=None) -> list:
+    columns = [
+        Host.id,
+        Host.account,
+        Host.org_id,
+        Host.display_name,
+        Host.ansible_host,
+        Host.facts,
+        Host.reporter,
+        Host.groups,
+        Host.system_profile_facts,
+        Host.modified_on,
+        Host.created_on,
+        Host.canonical_facts,
+        Host.per_reporter_staleness,
+        Host.groups,
+    ]
+
+    staleness_timestamps = Timestamps.from_config(inventory_config())
+    staleness = get_staleness_obj(identity)
+    filters = query_filters(rbac_filter=rbac_filter)
```

Comment:
To fix the current issue, we should do something like this

```suggestion
    filters, _= query_filters(rbac_filter=rbac_filter)
```